Libs/DICOM/Core/ctkDICOMDatabase.h

```diff
@@ -163,10 +163,18 @@ class CTK_DICOM_CORE_EXPORT ctkDICOMDatabase : public QObject
   Q_INVOKABLE QStringList patients();
   Q_INVOKABLE QStringList studiesForPatient(const QString patientUID);
   Q_INVOKABLE QStringList seriesForStudy(const QString studyUID);
-  Q_INVOKABLE QStringList instancesForSeries(const QString seriesUID);
+  /// Since a series may consists of many hundreds of instances, this method may be slow.
+  /// If hits > 0 is specified then returned instances will be limited to that number.
+  /// This is useful for retrieving first file, for example for getting access to fields within that file
+  /// using instanceValue() method.
+  Q_INVOKABLE QStringList instancesForSeries(const QString seriesUID, int hits = -1);
```

Comment:
Wouldn't we want to keep `instanceForSeries` with with a body that calls `instancesForSeries` with `hits == 1` for backward compatibility?

Reply:
This is backward compatible, because if someone does not specify `hits` then the default `-1` value means that all instances are returned (same as before).

The API follows Qt conventions (`hits` argument name, `-1` corresponding to return all matching values), see for example [here](https://doc.qt.io/archives/qt-4.8/qabstractitemmodel.html#match).

Reply:
Okay, I see, sorry, I had thought this was changing the name of the method from singular to plural but I see now it was always plural.